app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -182,6 +182,20 @@ open class HomeActivity : LocaleAwareAppCompatActivity() {
         BrowsersCache.resetAll()
     }
 
+    override fun onStart() {
+        super.onStart()
+
+        with(application.components.wifiIntegration) {
```

Comment:
What do you think about to put it on [BaseBrowserFragment](https://github.com/Baron-Severin/fenix/blob/2db2d9ceae0c43e933050555d60d67c5667b4d00/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt#L86)?  [As it behave the same way as other features that are related to the browser mode](https://github.com/Baron-Severin/fenix/blob/2db2d9ceae0c43e933050555d60d67c5667b4d00/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt#L91), it should be only active while the user is on the browser mode.

You could create something similar to [ToolbarIntegration](https://github.com/Baron-Severin/fenix/blob/master/app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt) as you are decorating an existing component with a new behaviour:

You could rename `WifiIntegration` to `SitePermissionsIntegration`

```kotlin
class SitePermissionsIntegration(
    private val settings: Settings,
    private val wifiConnectionMonitor: WifiConnectionMonitor
): LifecycleAwareFeature {

    /**
     * Adds listener for autplay setting [AUTOPLAY_ALLOW_ON_WIFI]. Sets all autoplay to allowed when
     * WIFI is connected, blocked otherwise.
     */
    private val wifiConnectedListener: ((Boolean) -> Unit) by lazy {
        { connected: Boolean ->
            val setting =
                if (connected) SitePermissionsRules.Action.ALLOWED else SitePermissionsRules.Action.BLOCKED
            settings.setSitePermissionsPhoneFeatureAction(PhoneFeature.AUTOPLAY_AUDIBLE, setting)
            settings.setSitePermissionsPhoneFeatureAction(PhoneFeature.AUTOPLAY_INAUDIBLE, setting)
        }
    }

    /**
     * If autoplay is only enabled on WIFI, sets a WIFI listener to set them accordingly. Otherwise
     * noop.
     */
   internal fun maybeAddWifiConnectedListener() {
        if (settings.getAutoplayUserSetting(default = AUTOPLAY_BLOCK_ALL) == AUTOPLAY_ALLOW_ON_WIFI) {
            addWifiConnectedListener()
        }
    }

    fun addWifiConnectedListener() {
        wifiConnectionMonitor.addOnWifiConnectedChangedListener(wifiConnectedListener)
    }

    internal fun removeWifiConnectedListener() {
        wifiConnectionMonitor.removeOnWifiConnectedChangedListener(wifiConnectedListener)
    }

    override fun start() {
        wifiConnectionMonitor.start()
        maybeAddWifiConnectedListener()
    }

    override fun stop() {
        removeWifiConnectedListener()
        wifiConnectionMonitor.stop()
    }
}
```
 You could init it in [BaseBrowserFragment](https://github.com/Baron-Severin/fenix/blob/2db2d9ceae0c43e933050555d60d67c5667b4d00/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt#L86) like similar integrations 

```kotlin

    private val sitePermissionsIntegration = ViewBoundFeatureWrapper<SitePermissionsIntegration>()

    protected open fun initializeUI(view: View): Session? {
            // Other initializations
 
            sitePermissionsIntegration.set(
                feature= SitePermissionsIntegration(
                    requireContext().settings(),
                    requireContext().wifiConnectionMonitor
                ),
                owner = this,
                view = view
            )

}
```

Reply:
Right now only browser stuff cares about the WIFI state, but it doesn't seem to me like there's a natural connection there.  We could easily want other things to listen to WIFI state across the app, which is why I put it in the activity.  I think there's an argument to be made for not overgeneralizing, but this still feels more appropriate to me.  What do you think?